src/thing_creature.c

```diff
@@ -1733,8 +1739,8 @@ void creature_cast_spell(struct Thing *castng, long spl_idx, long shot_lvl, long
         cctrl->teleport_x = trg_x;
         cctrl->teleport_y = trg_y;
     }
-    // Check if the spell can be fired as a shot
-    if (spconf->shot_model > 0)
+    // Check if the spell can be fired as a shot. It is definitely not if casted on itself.
+    if (spconf->shot_model > 0 && cctrl->targtng_idx != castng->index)
```

Comment:
in lines like this it is best to use extra brackets for readability:
` if ((spconf->shot_model > 0) && (cctrl->targtng_idx != castng->index))`

Reply:
OK.